tree/ntuple/v7/test/ntuple_types.cxx

```diff
@@ -210,6 +210,60 @@ TEST(RNTuple, StdTuple)
    }
 }
 
+TEST(RNTuple, StdSet)
+{
+   auto field = RField<std::set<int64_t>>("setField");
+   EXPECT_STREQ("std::set<std::int64_t>", field.GetType().c_str());
+   auto otherField = RFieldBase::Create("test", "std::set<int64_t>").Unwrap();
+   EXPECT_STREQ(field.GetType().c_str(), otherField->GetType().c_str());
+   EXPECT_EQ((sizeof(std::set<int64_t>)), field.GetValueSize());
+   EXPECT_EQ((sizeof(std::set<int64_t>)), otherField->GetValueSize());
+   EXPECT_EQ((alignof(std::set<int64_t>)), field.GetAlignment());
+   EXPECT_EQ((alignof(std::set<int64_t>)), otherField->GetAlignment());
+   auto setSetField = RField<std::set<std::set<CustomStruct>>>("setSetField");
+   EXPECT_STREQ("std::set<std::set<CustomStruct>>", setSetField.GetType().c_str());
+
+   FileRaii fileGuard("test_ntuple_rfield_stdset.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto set_field = model->MakeField<std::set<float>>({"mySet", "float set"});
+      // For templated set fields, no dictionary should be necessary.
+      auto set_field2 = model->MakeField<std::set<std::set<int>>>({"mySet2"});
```

Comment:
We should also test a set of a composite object (eg. `std::set<std::pair<float, double>>` (or a custom struct.